src/git.rs

```diff
@@ -0,0 +1,300 @@
+    pub message: Option<String>,
+    pub remote_url: Option<String>,
+}
+
+type GraphPushContextInput = graph::push::push_schema_mutation::GitContextInput;
+impl Into<GraphPushContextInput> for GitContext {
+    fn into(self) -> GraphPushContextInput {
+        GraphPushContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type GraphCheckContextInput = graph::check::check_schema_query::GitContextInput;
+impl Into<GraphCheckContextInput> for GitContext {
+    fn into(self) -> GraphCheckContextInput {
+        GraphCheckContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type SubgraphPushContextInput = subgraph::push::push_partial_schema_mutation::GitContextInput;
+impl Into<SubgraphPushContextInput> for GitContext {
+    fn into(self) -> SubgraphPushContextInput {
+        SubgraphPushContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+type SubgraphCheckContextInput = subgraph::check::check_partial_schema_query::GitContextInput;
+impl Into<SubgraphCheckContextInput> for GitContext {
+    fn into(self) -> SubgraphCheckContextInput {
+        SubgraphCheckContextInput {
+            branch: self.branch,
+            commit: self.commit,
+            committer: self.committer,
+            remote_url: self.remote_url,
+            message: self.message,
+        }
+    }
+}
+
+impl GitContext {
+    pub fn new(
+        override_branch: Option<String>,
+        override_committer: Option<String>,
+        override_commit: Option<String>,
+        override_remote_url: Option<String>,
+    ) -> Self {
+        let git = git_info::get();
+
+        let branch = if override_branch.is_none() {
+            git.current_branch
+        } else {
+            override_branch
+        };
+        let commit = if override_commit.is_none() {
+            git.head.last_commit_hash_short
+        } else {
+            override_commit
+        };
+
+        let mut remote_url = override_remote_url;
+        let mut committer = override_committer;
+
+        // if both remote_url and committer have values, we don't need to
+        // worry about executing this block
+        if remote_url.is_none() || committer.is_none() {
+            if let Some(mut config) = git.config {
+                // use the local git remote url if not provided in env var
+                remote_url = if remote_url.is_none() {
+                    config.remove("remote.origin.url")
+                } else {
+                    remote_url
+                };
+
```

Comment:
this block is a bit confusing to me - calling `config.remove` returns a value that we then use? is there a `get` method we could use instead?

Reply:
Me and irina were looking at this, and I think this is all we have. It's because we need ownership of the returned value. Everything else we could find returns a borrowed value. I'll add a comment